### DISCUSSION TITLE
Why changing the 'image of a label's transparency with CTkSlider doesn't work?

### DISCUSSION THREAD

**Discussion by duruburak** [TomSchimansky/CustomTkinter#1456] [General]
I'm trying to change the image's alpha value (transparency) with real-time data comes from `CTkSlider`. Here's my approach, what is wrong, do you guys think?

```python
import customtkinter
from PIL import Image

def change_alpha(value):
    global image, image_obj, label

    image_obj = image_obj.putalpha(int(value))
    image.configure(dark_image=image_obj)
    label.configure(image=image)

root = customtkinter.CTk()

image_obj = Image.open("image.png")
image = customtkinter.CTkImage(dark_image=image_obj, light_image=image_obj, size=(70,70))

label = customtkinter.CTkLabel(root, text="", image=image)
label.pack(pady=30, padx=20)

slider = customtkinter.CTkSlider(root, from_=0, to=255, command=change_alpha)
slider.pack(pady=30, padx=20)

root.mainloop()
```
Traceback:
```python
Exception in Tkinter callback
Traceback (most recent call last):
  File "D:\Python311\Lib\tkinter\__init__.py", line 1948, in __call__
    return self.func(*args)
           ^^^^^^^^^^^^^^^^
  File "D:\Python311\Lib\site-packages\customtkinter\windows\widgets\ctk_slider.py", line 324, in _clicked
    self._command(self._output_value)
  File "main.py", line 7, in change_alpha
    image_obj = image_obj.putalpha(int(value))
                ^^^^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'putalpha'
```

**Comment by AkashS10** [comment]
```python3
import customtkinter
from PIL import Image

def change_alpha(value):
    global image, new_image_obj, label

    new_image_obj = image_obj.putalpha(int(value))
    image.configure(dark_image=new_image_obj)
    label.configure(image=image)

root = customtkinter.CTk()

image_obj = Image.open("image.png")
image = customtkinter.CTkImage(dark_image=image_obj, light_image=image_obj, size=(70,70))

label = customtkinter.CTkLabel(root, text="", image=image)
label.pack(pady=30, padx=20)

slider = customtkinter.CTkSlider(root, from_=0, to=255, command=change_alpha)
slider.pack(pady=30, padx=20)

root.mainloop()
```

I think that by making the `image_obj` global, you are overwriting it with a blank value. You should definitely use classes for instead of using global in functions. But I think that this code might work. I haven't tested it either, so let me know if it works

**Comment by duruburak** [comment]
> ```python
> import customtkinter
> from PIL import Image
> 
> def change_alpha(value):
>     global image, new_image_obj, label
> 
>     new_image_obj = image_obj.putalpha(int(value))
>     image.configure(dark_image=new_image_obj)
>     label.configure(image=image)
> 
> root = customtkinter.CTk()
> 
> image_obj = Image.open("image.png")
> image = customtkinter.CTkImage(dark_image=image_obj, light_image=image_obj, size=(70,70))
> 
> label = customtkinter.CTkLabel(root, text="", image=image)
> label.pack(pady=30, padx=20)
> 
> slider = customtkinter.CTkSlider(root, from_=0, to=255, command=change_alpha)
> slider.pack(pady=30, padx=20)
> 
> root.mainloop()
> ```
> 
> I think that by making the `image_obj` global, you are overwriting it with a blank value. You should definitely use classes for instead of using global in functions. But I think that this code might work. I haven't tested it either, so let me know if it works

No, this didn't work. I'm actually using classes, I just created a small reproducible example out of it.
